### PR TITLE
Update doc with current status

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Updat COM document with current state of agent

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]

File: docs/dev/caveats.md [hunk 1]
```diff
@@ -4,19 +4,24 @@ This document provides a list of known development caveats
 
 ## Windows
 
+### COM Caveats
+
 The COM concurrency model may be set in different ways, it also has to be called for every thread that might indeed interact with the COM library. Furthermore, once a concurrency model is set for a thread, it cannot be changed unless the thread is `CoUnitilialize()`d. This poses an issue for us for a variety of reasons:
-1. We use thirdparty libraries like `gopsutil` that initialize the concurrency model setting it to the multi-threaded model - the library will fail in its calls if the model is any different.
+1. Third party libraries like `gopsutil` that initialize the concurrency model setting it to the multi-threaded model - the library will fail in its calls if the model is any different.
```
Comment (sarina-dd) [inline docs/dev/caveats.md:10]:
```suggestion
1. Third party libraries like `gopsutil`, which initializes the concurrency model by setting it to the multi-threaded model, will fail in its calls as the model is different.
```

File: docs/dev/caveats.md [hunk 2]
```diff
@@ -4,19 +4,24 @@ This document provides a list of known development caveats
...
-and involved go checks and facilities (CPU check, which calls `gopsutil` to collect CPU information during its configure phase) may be set up successfully. 
 
-Once the agent is finally up, and we get past the check setup, no additional COM calls will currently be made from go-land. However, we do continue to make these calls from python checks. Python checks will set/use the concurrency model as they please (typically the single-threaded model). This has a few implications:
+
+To support python checks which use COM, we call  `CoInitializeEx(0)` in the python loader while checks are getting loaded, and running `CoUninitialize()` immediately after loading. By doing so, when `pythoncom` is imported during the loading of checks the concurrency model is already set.
```
Comment (sarina-dd) [inline docs/dev/caveats.md:16]:
```suggestion
To support Python checks which use COM, call  `CoInitializeEx(0)` in the python loader while checks are loading, and run `CoUninitialize()` immediately after loading. By doing so, `pythoncom` is imported during the loading of checks and the concurrency model is already set.
```

File: docs/dev/caveats.md [hunk 3]
```diff
@@ -4,19 +4,24 @@ This document provides a list of known development caveats
...
+### WMI
+WMI is implemented via COM.  Therefore, all of the above apply to any code that directly or indirectly uses WMI.  We have removed _all_ use of WMI in the core agent, and in fact have [removed](https://github.com/DataDog/datadog-agent/blob/master/tasks/go.py#L295-L299) the `gopsutil` dependency to ensure that we don't inadvertently reintroduce this dependency.
```
Comment (sarina-dd) [inline docs/dev/caveats.md:18]:
```suggestion
WMI is implemented via COM. Therefore, all of the above apply to any code that directly or indirectly uses WMI.  _All_ use of WMI in the core agent is removed, along with the [removal](https://github.com/DataDog/datadog-agent/blob/master/tasks/go.py#L295-L299) of the `gopsutil` dependency to ensure that this dependency is not reintroduced.
```

File: docs/dev/caveats.md [hunk 4]
```diff
@@ -4,19 +4,24 @@ This document provides a list of known development caveats
...
+
+
+However, we do continue to make these calls from python checks. Python checks will set/use the concurrency model as they please (typically the single-threaded model). This has a few implications:
```
Comment (sarina-dd) [inline docs/dev/caveats.md:21]:
```suggestion
However, Python checks do continue to make these calls. Python checks will set/use the concurrency model as they please (typically the single-threaded model). This has a few implications:
```

File: docs/dev/caveats.md [hunk 5]
```diff
@@ -4,19 +4,24 @@ This document provides a list of known development caveats
...
 - We cannot assume anything about the concurrency model of a thread after checks are loaded. 
-- Any call to a `gopsutil` that might rely on WMI is liable to failure as it might try to initialize on a multi-threaded model while the thread might be on the single-threaded one.  
+- Any call to a library that might rely on WMI is liable to fail as it might try to initialize on a multi-threaded model while the thread might be on the single-threaded one.  
 - Since we call `CoInitializeEx(0)` when loading checks, the current behavior would break Auto-Discovery and dynamic reload of checks on windows.
 
-
+WMI in general is not very stable. A great deal of effort has been made to remove WMI from the core agent.  Additional effort has been made to remove WMI from the python checks.  Take care to not (re)introduce WMI into the agent.
```
Comment (sarina-dd) [inline docs/dev/caveats.md:26]:
```suggestion
WMI in general is not very stable. A great deal of effort has been made to remove WMI from the core agent.  Additional effort has been made to remove WMI from the Python checks. Take care to not (re)introduce WMI into the Agent.
```

**Review by sarina-dd — COMMENTED** [review]
Docs suggestions

**Pull request merged**